### PR TITLE
plausible, nixos/plausible: Add `listenAddress` option

### PR DESCRIPTION
**Pull request by nh2**
Replaces PR #130297 based on what we learned in there (https://github.com/NixOS/nixpkgs/pull/130297#issuecomment-1805851333).

See commit message for more details.

---

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

CC #130297 contributors:

@happysalada @RaitoBezarius @softinio @Ma27 @SuperSandro2000

### CODE REVIEW

**Comment by nh2** [comment]
CC @azahi @Tom-Hubrecht from https://github.com/NixOS/nixpkgs/pull/130297#issuecomment-1806005444

File: nixos/doc/manual/release-notes/rl-2311.section.md [hunk 1]
```diff
@@ -317,6 +317,8 @@
 
 - Package `pash` was removed due to being archived upstream. Use `powershell` as an alternative.
 
+- The option `services.plausible.releaseCookiePath` has been removed: Plausible does not use any distributed Erlang features, and does not plan to (see [discussion](https://github.com/NixOS/nixpkgs/pull/130297#issuecomment-1805851333)), so NixOS now disable them, and the Erlang cookie becomes unnecessary. You may delete the file that `releaseCookiePath` was set to.
```
Comment (happysalada) [inline nixos/doc/manual/release-notes/rl-2311.section.md:320]:
```suggestion
- The option `services.plausible.releaseCookiePath` has been removed: Plausible does not use any distributed Erlang features, and does not plan to (see [discussion](https://github.com/NixOS/nixpkgs/pull/130297#issuecomment-1805851333)), so NixOS now disables them, and the Erlang cookie becomes unnecessary. You may delete the file that `releaseCookiePath` was set to.
```

**Review by happysalada — APPROVED** [review]

**Comment by happysalada** [comment]
This looks good to me, just a typo. I havent tested though, it would be nice to have a couple of different people test this.

File: nixos/doc/manual/release-notes/rl-2311.section.md (hunk 1 above)
Comment (nh2) [reply nixos/doc/manual/release-notes/rl-2311.section.md]: Thanks, fixed!

**Comment by nh2** [comment]
What I tested:

* The NixOS tests
* Checked the the landing page looks correctly in a throwaway VM via `QEMU_NET_OPTS` (also improved the manual for it, see #266709)
* Deployed it to my production Plausible (1.4, not 2.0) backported to NixOS 23.05

**Comment by happysalada** [comment]
@softinio any chance you have a bit of time to test the 2.0 release ? Just to be safe

File: nixos/modules/services/web-apps/plausible.nix [hunk 2]
```diff
@@ -180,7 +180,8 @@ in {
       enable = true;
     };
 
-    services.epmd.enable = true;
+    # See note [plausible-needs-no-erlang-distributed-features].
+    services.epmd.enable = false;
```
Comment (Tom-Hubrecht) [inline nixos/modules/services/web-apps/plausible.nix:184]: The default for `services.epmd` is false, so just unsetting the option would do what we want, setting it explicitely to `false` might clash with another service using it and breaking some setups.

**Comment by Tom-Hubrecht** [comment]
Otherwise, I tested it on 2.0 and it works fine

File: nixos/modules/services/web-apps/plausible.nix [hunk 3]
```diff
@@ -11,13 +11,6 @@ in {
 
     package = mkPackageOptionMD pkgs "plausible" { };
 
-    releaseCookiePath = mkOption {
```
Comment (azahi) [inline nixos/modules/services/web-apps/plausible.nix:14]: Should we add `mkRemovedOptionModule` for this option?

**Review by azahi — COMMENTED** [review]
I've tested it out with my Plausible deployment and it works great.

File: nixos/modules/services/web-apps/plausible.nix (hunk 3 above)
Comment (nh2) [reply nixos/modules/services/web-apps/plausible.nix:14]: Good idea, done.

File: nixos/modules/services/web-apps/plausible.nix (hunk 2 above)
Comment (nh2) [reply nixos/modules/services/web-apps/plausible.nix]: Done.

**Pull request merged**